src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PerformanceAnalyzerController.java

```diff
@@ -214,4 +269,52 @@ private void saveStateToConf(boolean featureEnabled, String fileName) {
             }
         });
     }
+
+    private synchronized void saveMutedRcasToConf(String mutedRcas, String fileName) {
+        try {
+            // create the config json Object from rca config file
+            Path rcaConfPath = Paths.get(getPAConfigDirectory().toString(), fileName);
+            Scanner scanner = new Scanner(new FileInputStream(rcaConfPath.toString()), StandardCharsets.UTF_8.name());
+            String jsonText = scanner.useDelimiter("\\A").next();
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.enable(JsonParser.Feature.ALLOW_COMMENTS);
+            mapper.enable(SerializationFeature.INDENT_OUTPUT);
+            JsonNode configObject = mapper.readTree(jsonText);
+
+            // To ensure consistency on read for rca.conf and back-up purpose, we will maintain 2 files,
+            // the (n) 'rca.conf' and the (n-1) 'rca_<timestamp_n1>.conf'. For updating the latest config, we will :
+            //
+            // 1. Back-up the current (n) 'rca.conf' as (n-1) 'rca_<timestamp_n1>.conf'
+            // 2. Update the config object, write to a temp file and atomically move the temp file to rca.conf
+            // 3. Delete the older (n-2) 'rca_<timestamp_n2>.conf' file to ensure we strictly maintain 2 files.
+            String seperator = "_";
+
+            String currTimeStamp = new SimpleDateFormat("yyyyMMddHHmm").format(new Date());
+            Path backupRcaConfPath = Paths.get(getPAConfigDirectory().toString(), fileName + seperator + currTimeStamp);
+            Files.copy(rcaConfPath, backupRcaConfPath);
+            LOG.debug("Created the back-up Conf file: {} for : {}", backupRcaConfPath.toString(), rcaConfPath.toString());
+
+            ((ObjectNode) configObject).put(MUTED_RCAS_CONFIG, mutedRcas);
+            Path tmp = Files.createTempFile(getPAConfigDirectory(), null, null);
+            mapper.writeValue(new FileOutputStream(tmp.toString()), configObject);
+            Files.move(tmp, rcaConfPath, StandardCopyOption.ATOMIC_MOVE);
+            LOG.debug("Updated the Conf File: {} for Muted RCAs: {}", rcaConfPath.toString(), mutedRcas);
+
+            // Get all files which begin with filename_, for example : rca.conf_202004291057 for filename 'rca.conf'
+            File rootFolder = new File(getPAConfigDirectory().toString());
+            String[] targetFiles = rootFolder.list((path, name) -> name.startsWith(fileName + seperator));
```

Comment:
This will kill us if we have a lot of files. I would just suggest calling the old rca.conf file as `rca.conf.0`. That way we don't care about deleting. We just overwrite.

Reply:
Make sure that the overwrites and not fails if the file already exists